sale_product_approval/tests/test_product_approval.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestSaleOrderLineDates(TransactionCase):
```

Comment:
SavepointCase (until v14) is better in terms of performance.

You can look at this at 6:22:00 https://www.youtube.com/watch?v=lT___hG110Q